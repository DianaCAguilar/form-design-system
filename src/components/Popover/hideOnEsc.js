const hideOnEscFunc = (callback) => ({
  name: 'hideOnEsc',
  defaultValue: true,
  fn(props) {
    function onKeyDown(event) {
      if (event.keyCode === 27) {
        if (callback) {
          callback();
        } else {
          props.hide();
        }
      }
    }

    return {
      onShow() {
        document.addEventListener('keydown', onKeyDown);
      },
      onHide() {
        document.removeEventListener('keydown', onKeyDown);
      },
    };
  },
});

export default hideOnEscFunc;
