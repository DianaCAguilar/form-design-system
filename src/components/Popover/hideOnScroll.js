const hideOnScroll = (closeOnScrollRef) => ({
  name: 'hideOnEsc',
  defaultValue: true,
  fn({ hide }) {
    if (!closeOnScrollRef) return {};

    function scrollLogic() {
      hide();
      closeOnScrollRef.current.removeEventListener('scroll', scrollLogic);
    }

    return {
      onShow() {
        closeOnScrollRef.current.addEventListener('scroll', scrollLogic);
      },
      onHide() {
        closeOnScrollRef.current.removeEventListener('scroll', scrollLogic);
      },
    };
  },
});

export default hideOnScroll;
