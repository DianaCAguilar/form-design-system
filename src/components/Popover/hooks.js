import { useEffect } from 'react';

let overflowStyle = null;

/**
 * Hook that disables scroll on a DOM node when `isDisabled` is true.
 *
 * @param {Object} disableScrollRef - react ref to DOM node
 * @param {Boolean} isDisabled
 */
export const useDisableScroll = (disableScrollRef, isDisabled) => {
  useEffect(() => {
    if (disableScrollRef) {
      const domNode = disableScrollRef.current || disableScrollRef;
      if (isDisabled) {
        overflowStyle = domNode.style.overflow;
        domNode.style.overflow = 'hidden';
      } else if (overflowStyle !== null) {
        domNode.style.overflow = overflowStyle;
      }
    }
  }, [disableScrollRef, isDisabled]);
};

/**
 * Hook that invokes `closeCallback` if a popover is active when a user scrolls.
 *
 * @param {Object} closeOnScrollRef - react ref to scrolling DOM node
 * @param {Boolean} isActive - if the popover is currently open/active
 * @param {Boolean} closeCallback - function that closes the popover
 */
export const useCloseOnScroll = (closeOnScrollRef, isActive, closeCallback) => {
  useEffect(() => {
    if (closeOnScrollRef && isActive) {
      const scrollRef = closeOnScrollRef.current || closeOnScrollRef;
      scrollRef.addEventListener('scroll', function scrollLogic() {
        closeCallback();
        scrollRef.removeEventListener('scroll', scrollLogic);
      });
    }
  }, [closeOnScrollRef, isActive]);
};
