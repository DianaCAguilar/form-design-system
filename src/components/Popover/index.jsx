import React, { useMemo } from 'react';
import PropTypes from 'prop-types';
import Tippy from '@tippyjs/react';
import FDS from 'lib/dictionary/js/styleConstants';
import hideOnEscFunc from './hideOnEsc';

const Popover = ({
  children,
  isOpen,
  placement = 'bottom-start',
  trigger = 'click',
  triggerElement,
  onUserDismiss,
  theme = 'default',
  appendTo = () => document.body,
}) => {
  const hideOnEsc = useMemo(() => hideOnEscFunc(onUserDismiss), [onUserDismiss]);
  /* Tippy will throw a warning if you attempt to pass a trigger when
     isOpen is passed. This will avoid showing that warning. */
  const triggerHelper = isOpen !== undefined ? undefined : trigger;
  return (
    <>
      <Tippy
        animation={false}
        arrow={false}
        visible={isOpen}
        theme={`popover-${theme}`}
        trigger={triggerHelper}
        placement={placement}
        interactive
        appendTo={appendTo}
        onUserDismiss={onUserDismiss || (() => {})}
        plugins={[hideOnEsc]}
        ignoreAttributes
        zIndex={FDS.ZINDEX_MODAL}
        moveTransition="transform 0.2s ease-out"
        content={children}
      >
        {triggerElement}
      </Tippy>
    </>
  );
};

Popover.propTypes = {
  appendTo: PropTypes.oneOfType([
    PropTypes.node,
    PropTypes.func,
    PropTypes.oneOf(['parent']),
  ]),
  children: PropTypes.node,
  /** Element that popover is positioned on */
  triggerElement: PropTypes.node,
  /** Position preference of the popover */
  placement: PropTypes.oneOf(['auto', 'top', 'right', 'bottom', 'bottom-start', 'left']),
  /** Controls the popover */
  isOpen: PropTypes.bool,
  /** calls this callback when user clicks outside popover or presses esc key */
  onUserDismiss: PropTypes.func,
  /** What mode of interaction triggers the popover */
  trigger: PropTypes.oneOf(['click', 'mouseenter focus']),
  /** the UI display */
  theme: PropTypes.oneOf(['default', 'unthemed']),
};

export default Popover;
