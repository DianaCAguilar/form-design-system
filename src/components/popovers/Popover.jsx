import React, { useRef, useEffect, useState } from 'react';
import ReactDOM from 'react-dom';
import PropTypes from 'prop-types';
import { Manager, Reference, Popper } from 'react-popper';
import { isNotRefsEvent } from '../util/events';

export const VALID_POSITIONS = ['auto', 'top', 'right', 'bottom', 'left'];
export const VALID_ALIGNMENTS = ['start', 'end', 'center'];
export const VALID_INTERACTION_MODES = ['hover', 'click', 'controlled'];

/**
 * https://popper.js.org/popper-documentation.html#Popper.placements
 *
 * @param {String} position prop value
 * @param {String} alignment prop value
 * @returns {String} valid `placement` value for PopperJS `Popper` component
 */
export const getPopperPlacement = (position, alignment) => {
  // Popperjs does not have a "center" placement variation. It centers by default.
  // Our component explicitly accepts a `center` value for alignment.
  const variation = !alignment || alignment === 'center' ? '' : `-${alignment}`;
  return `${position}${variation}`;
};

/**
 * @param {Object} props react props
 * @returns {ReactElement}
 */
const Popover = ({
  trigger,
  children,
  interactionMode,
  disablePortal,
  position,
  alignment,
  distance,
  isOpen,
}) => {
  const [isActive, setIsActive] = useState(false);
  const [isFocused, setIsFocused] = useState(false);
  const refTrigger = useRef(null);
  let refContent = null; // must be assigned via setter fn in `Popper`

  // update active state on props change to accommodate fully controlled popovers
  useEffect(() => {
    setIsActive(interactionMode === 'controlled' && isOpen);
  }, [interactionMode, isOpen]);

  /**
   * Closes popover when user presses ESC
   * @param {Event} e DOMEvent
   */
  const handleKeyPress = (e) => {
    const isEscapeKey = ['Esc', 'Escape'].some((key) => key === e.key);
    if (isEscapeKey) setIsActive(false);
  };

  /**
   * Closes popover when user clicks outside of content or trigger
   * @param {Event} e DOMEvent
   */
  const handleBodyMouseMove = (e) => {
    const isNotPopoverHover = isNotRefsEvent([refTrigger, refContent], e);
    if (!isFocused && isNotPopoverHover) setIsActive(false);
  };

  /**
   * Closes popover when user clicks outside of content or trigger
   * @param {Event} e DOMEvent
   */
  const handleBodyClick = (e) => {
    const isNotPopoverClick = isNotRefsEvent([refTrigger, refContent], e);
    if (isNotPopoverClick) setIsActive(false);
  };

  useEffect(() => {
    /* eslint-disable no-undef */
    document.body.addEventListener('mousedown', handleBodyClick, false);
    document.body.addEventListener('keyup', handleKeyPress, false);

    if (interactionMode === 'hover') {
      document.body.addEventListener('mousemove', handleBodyMouseMove, false);
    }

    return () => {
      document.body.removeEventListener('mousedown', handleBodyClick, false);
      document.body.removeEventListener('keyup', handleKeyPress, false);

      if (interactionMode === 'hover') {
        document.body.removeEventListener('mousemove', handleBodyMouseMove, false);
      }
    };
    /* eslint-enable no-undef */
  });

  let triggerProps = {};
  switch (interactionMode) {
    case 'hover':
      triggerProps.onMouseEnter = () => {
        setIsActive(true);
      };
      triggerProps.onFocus = () => {
        setIsActive(true);
        setIsFocused(true);
      };
      triggerProps.onBlur = () => {
        setIsActive(false);
        setIsFocused(false);
      };
      triggerProps.tabIndex = '1';
      break;
    case 'click':
      triggerProps.onClick = () => {
        setIsActive(!isActive);
      };
      break;
    default:
      triggerProps = {};
  }

  // https://popper.js.org/popper-documentation.html#modifiers
  const popperModifiers = {
    offset: {
      enabled: interactionMode !== 'hover',
      offset: `0,${distance}`,
    },
  };

  const PopperContent = () => (
    <Popper
      innerRef={(node) => {
        refContent = node;
      }}
      modifiers={popperModifiers}
      placement={getPopperPlacement(position, alignment)}
    >
      {({ placement, ref, style }) => (
        <div ref={ref} style={style} data-placement={placement}>
          {children}
        </div>
      )}
    </Popper>
  );

  return (
    <Manager>
      <Reference>
        {({ ref }) => (
          <div ref={ref} aria-haspopup="true" aria-expanded={isActive.toString()}>
            <div ref={refTrigger} {...triggerProps}>
              {trigger}
            </div>
          </div>
        )}
      </Reference>
      {isActive &&
        (disablePortal ? (
          <PopperContent />
        ) : (
          ReactDOM.createPortal(
            <PopperContent />,
            document.body /* eslint-disable-line no-undef */
          )
        ))}
    </Manager>
  );
};

Popover.defaultProps = {
  interactionMode: 'click',
  position: 'auto',
  alignment: 'start',
  distance: 4,
};

Popover.propTypes = {
  /** JSX - Acts as a positioning reference for the popover and triggers active state */
  trigger: PropTypes.oneOfType([PropTypes.node, PropTypes.element]).isRequired,

  /**
   * JSX - Content to place in the positioned popover container. The popover container
   * does not provide any default styling; content should be styled with background,
   * borders, and shadows as appropriate.
   */
  children: PropTypes.oneOfType([PropTypes.node, PropTypes.element]).isRequired,

  /**
   * What mode of interaction triggers the active/inactive state of the popover.
   *
   * `hover` - popover opens on trigger hover/focus
   *
   * `click` - popover opens on trigger click
   *
   * `controlled` - enables "fully controlled" mode in which the popover is only active
   * when the `isActive` prop is set to `true`
   */
  interactionMode: PropTypes.oneOf(VALID_INTERACTION_MODES),

  /** Controlls active state of popover when in fully controlled interaction mode */
  isOpen: PropTypes.bool,

  /** disables portaling the popover to `document.body` */
  disablePortal: PropTypes.bool,

  /**
   * Places the popover content on the given side of the trigger.
   * `top` for example, will place the popover content above the trigger.
   */
  position: PropTypes.oneOf(VALID_POSITIONS),

  /** Controls alignment of popover content relative to trigger */
  alignment: PropTypes.oneOf(VALID_ALIGNMENTS),

  /** Offset distance from trigger. Ignored in 'hover' interaction mode. */
  distance: PropTypes.number,
};

export default Popover;
