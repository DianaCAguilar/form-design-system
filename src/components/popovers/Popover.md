# Popover

Base popover component. Bring your own trigger and popover content.

## What this component does

- Handle events according to `interactionMode` passed in
- Handle positioning of popover content

## What this component does **not** do

- Content styling - the popover container is just a transparent div.
- Arrows. They aren't supported, but we can add them in the future.
