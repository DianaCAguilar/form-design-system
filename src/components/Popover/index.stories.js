import React, { useState } from 'react';
import Button from 'components/Button';
import Popover from '.';

const Template = (args) => <Popover {...args} />;
export const Primary = Template.bind({});
Primary.args = {
  triggerElement: <Button label="Click Me" />,
  children: <div className="padding--all">popover content</div>,
};

export const Unthemed = Template.bind({});
Unthemed.args = {
  theme: 'unthemed',
  triggerElement: <Button label="Click Me" />,
  children: (
    <div
      className="bgColor-white"
      style={{ border: 'red dotted 3px', background: 'white', padding: '8px' }}
    >
      popover content
    </div>
  ),
};

export const controlledPopover = (args) => {
  const [isOpen, setIsOpen] = useState(true);
  return (
    <Popover
      {...args}
      onUserDismiss={() => setIsOpen(false)}
      triggerElement={
        <button
          onClick={() => {
            setIsOpen(!isOpen);
          }}
        >
          Open popover
        </button>
      }
      isOpen={isOpen}
    >
      <div className="padding--all">
        <h3 className="type--head3">Look at me</h3>
        <p>
          <em>i am the popover</em>
        </p>
      </div>
    </Popover>
  );
};

export default {
  component: Popover,
  title: 'components/Popover',
};
