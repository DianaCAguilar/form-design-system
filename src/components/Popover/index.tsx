import * as RadixPopover from '@radix-ui/react-popover';
import React, { useEffect, useState, Ref, RefObject, HTMLAttributes } from 'react';
import { CSSTransition } from 'react-transition-group';
import FDS from 'dictionary/js/styleConstants';
import { useDisableScroll, useCloseOnScroll } from './hooks';

export type Position = 'top' | 'right' | 'bottom' | 'left';
export type InteractionMode = 'hover' | 'click' | 'controlled';

export interface PopoverProps {
  /** JSX - Acts as a positioning reference for the popover and triggers active state */
  trigger: React.ReactElement;

  /**
   * JSX - Content to place in the positioned popover container. The popover container
   * does not provide any default styling; content should be styled with background,
   * borders, and shadows as appropriate.
   */
  children: React.ReactNode | React.ReactElement;

  /**
   * What mode of interaction triggers the active/inactive state of the popover.
   *
   * `hover` - popover opens on trigger hover/focus
   *
   * `click` - popover opens on trigger click
   *
   * `controlled` - enables "fully controlled" mode in which the popover is only active
   * when the `isOpen` prop is set to `true`. Use `onUserDismiss` to handle dismissal
   * events from user interaction.
   */
  interactionMode?: InteractionMode;

  /** Controlls active state of popover when in fully controlled interaction mode */
  isOpen?: boolean;

  /**
   * Callback fired when user takes an action to dismiss the popover.
   * (e.g. ESC press, clicking outside, etc.)
   * Useful for updating `isOpen` in controlled mode.
   * Function is invoked with the DOM event as the argument.
   */
  onUserDismiss?: (e: MouseEvent | KeyboardEvent) => void;

  /** disables portaling the popover to `document.body` */
  disablePortal?: boolean;

  /**
   * Vertical position preference of popover content.
   * `top` for example, will place the popover content above the trigger.
   */
  position?: Position;

  /** Horizontal alignment preference of popover content relative to trigger */
  alignment?: 'start' | 'center' | 'end';

  /** Offset distance from trigger. */
  distance?: number;

  /** Delay in milliseconds for popover to trigger (hover mode only) */
  delay?: number;

  /** Name of transition for popover content */
  transitionName?: 'GrowFast';

  /** Callback called when popover opens */
  onOpen?: () => void;

  /** Callback called when popover closes */
  onClose?: () => void;

  /** When popover opens, this element should freeze scrolling. When the popover closes, the scrollability of this element should resume. */
  disableScrollRef?: RefObject<HTMLElement>;

  /** When the Popover detects scroll events from this ref, the popover should close. */
  closeOnScrollRef?: RefObject<HTMLElement>;

  /** Ref to forward to the content container of the popover */
  ref?: Ref<HTMLElement>;
}

const Popover = ({
  interactionMode = 'click',
  disablePortal,
  position = 'bottom',
  alignment = 'start',
  distance = 4,
  delay = 0,
  onOpen = () => {},
  onClose = () => {},
  trigger,
  children,
  isOpen,
  transitionName,
  disableScrollRef,
  closeOnScrollRef,
}: PopoverProps): JSX.Element => {
  const [isActive, setIsActive] = useState(false);
  useCloseOnScroll(closeOnScrollRef, isActive, () => setIsActive(false));
  useDisableScroll(disableScrollRef, isActive);

  // update active state on props change to accommodate fully controlled popovers
  useEffect(() => {
    setIsActive(interactionMode === 'controlled' && !!isOpen);
  }, [interactionMode, isOpen]);

  let triggerProps: HTMLAttributes<HTMLElement> = {};
  let hoverTimeout: number;
  switch (interactionMode) {
    case 'hover':
      triggerProps.onMouseEnter = () => {
        if (delay > 0) {
          hoverTimeout = setTimeout(setIsActive, delay, true);
        } else {
          setIsActive(true);
        }
      };
      triggerProps.onMouseLeave = () => {
        if (hoverTimeout) {
          clearTimeout(hoverTimeout);
        }
        setIsActive(false);
      };
      triggerProps.onKeyUp = (e) => {
        if (e.key === 'Tab') {
          setIsActive(true);
        }
      };
      triggerProps.onKeyDown = (e) => {
        if (e.key === 'Tab') {
          setIsActive(false);
        }
      };
      triggerProps.onClick = (e) => {
        e.stopPropagation();
        setIsActive(false);
      };
      triggerProps.tabIndex = 1;
      break;
    case 'click':
      triggerProps.onClick = (e) => {
        e.stopPropagation();
        setIsActive(!isActive);
      };
      break;
    default:
      triggerProps = {};
  }

  const clonedTrigger = React.cloneElement(trigger, triggerProps);

  const contentStyle = {
    zIndex: Number(FDS.ZINDEX_MODAL),
  };

  const popperContent = (
    <CSSTransition
      onEntered={onOpen}
      onExited={onClose}
      in={isActive}
      unmountOnExit
      timeout={transitionName ? 200 : 0}
      classNames={transitionName ? `rtg${transitionName}` : undefined}
    >
      <div>{children}</div>
    </CSSTransition>
  );

  return (
    <RadixPopover.Root defaultOpen={isOpen}>
      <RadixPopover.Trigger asChild>{clonedTrigger}</RadixPopover.Trigger>
      <RadixPopover.Content
        style={{
          ...contentStyle,
        }}
        className="ease-in-out"
        align={alignment}
        side={position}
        sideOffset={distance}
        portalled={!!disablePortal}
      >
        {popperContent}
      </RadixPopover.Content>
    </RadixPopover.Root>
  );
};

Popover.displayName = 'Popover';

export default Popover;
