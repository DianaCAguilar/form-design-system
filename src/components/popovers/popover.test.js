import React from 'react';
import { shallow, mount } from 'enzyme';

import Popover, { getPopperPlacement } from './Popover';

const Trigger = () => (<button>trigger</button>);
const Content = () => (<p>popover content</p>);

const SELECTOR_TRIGGER = '[aria-haspopup] > div';
const SELECTOR_CONTENT = 'Popper'; // react-popper content displayName

/**
 * @param {ReactWrapper} w enzyme react wrapper
 * @returns {Boolean} if the popper exists in the DOM
 */
const isPopperOpen = (w) => w.exists(SELECTOR_CONTENT);


describe('Popover component', () => {

  it('matches snapshot (default props)', () => {
    const wrapper = mount(
      <Popover trigger={<Trigger />}>
        <Content />
      </Popover>
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('ignores `isOpen` prop when not in controlled mode', () => {
    const wrapper = shallow(
      <Popover
        trigger={<Trigger />}
        isOpen
      >
        <Content />
      </Popover>
    );
    expect(isPopperOpen(wrapper)).toBe(false);
  });

  describe('getPopperPlacement', () => {
    it('returns an alignment in the placement', () => {
      const actual = getPopperPlacement('bottom', 'start');
      const expected = 'bottom-start';
      expect(actual).toBe(expected);
    });
    it('does not return alignment when none is passed in', () => {
      const actual = getPopperPlacement('bottom');
      const expected = 'bottom';
      expect(actual).toBe(expected);
    });
    it('does not return placement alignment when "center" alignment is passed', () => {
      const actual = getPopperPlacement('bottom', 'center');
      const expected = 'bottom';
      expect(actual).toBe(expected);
    });
  });

  describe('interactionMode="click"', () => {
    let wrapper;
    let triggerEl;

    beforeEach(() => {
      wrapper = mount(
        <Popover trigger={<Trigger />} interactionMode="click">
          <Content />
        </Popover>
      );
      triggerEl = wrapper.find(SELECTOR_TRIGGER);
    });

    afterEach(() => {
      wrapper = null;
      triggerEl = null;
    });

    it('adds click-related props to trigger', () => {
      const triggerPropNames = Object.keys(triggerEl.props());
      expect(triggerPropNames).toContain('onClick');
    });

    it('opens and closes popover on trigger clicks', () => {
      expect(isPopperOpen(wrapper)).toBe(false);
      triggerEl.simulate('click');
      expect(isPopperOpen(wrapper)).toBe(true);
      triggerEl.simulate('click');
      expect(isPopperOpen(wrapper)).toBe(false);
    });

    it('closes on clicks away from popover', () => {
      expect(isPopperOpen(wrapper)).toBe(false);
      triggerEl.simulate('click');
      expect(isPopperOpen(wrapper)).toBe(true);
      triggerEl.simulate('click', { target: '<p>lol not the popover</p>' });
      expect(isPopperOpen(wrapper)).toBe(false);
    });
  });

  describe('interactionMode="hover"', () => {
    let wrapper;
    let triggerEl;

    beforeEach(() => {
      wrapper = mount(
        <Popover trigger={<Trigger />} interactionMode="hover">
          <Content />
        </Popover>
      );
      triggerEl = wrapper.find(SELECTOR_TRIGGER);
    });

    afterEach(() => {
      wrapper = null;
      triggerEl = null;
    });

    it('adds hover-related props to trigger', () => {
      const triggerPropNames = Object.keys(triggerEl.props());
      ['onMouseEnter', 'onFocus', 'onBlur', 'tabIndex'].forEach((prop) => {
        expect(triggerPropNames).toContain(prop);
      });
    });

    it('opens and closes popover on focus/blur', () => {
      expect(isPopperOpen(wrapper)).toBe(false);
      triggerEl.simulate('focus');
      expect(isPopperOpen(wrapper)).toBe(true);
      triggerEl.simulate('blur');
      expect(isPopperOpen(wrapper)).toBe(false);
    });

    it('opens on hover', () => {
      expect(isPopperOpen(wrapper)).toBe(false);
      triggerEl.simulate('mouseenter');
      expect(isPopperOpen(wrapper)).toBe(true);
    });

    it('closes on mousemove outside of Popover elements', () => {
      expect(isPopperOpen(wrapper)).toBe(false);
      triggerEl.simulate('mouseenter');
      expect(isPopperOpen(wrapper)).toBe(true);
      triggerEl.simulate('mousemove', { target: '<p>lol not the popover</p>' });
    });

  });

});
