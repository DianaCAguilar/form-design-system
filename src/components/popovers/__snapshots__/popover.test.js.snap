// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Popover component matches snapshot (default props) 1`] = `
<Manager>
  <Reference>
    <Component />
  </Reference>
</Manager>
`;

exports[`Popover component matches snapshot for disabled portal 1`] = `
<Manager>
  <Reference>
    <Component />
  </Reference>
</Manager>
`;
